File: LAB3/app/src/main/java/com/example/bmicalculator/DisplayMessageActivity.java
package com.example.bmicalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class DisplayMessageActivity extends AppCompatActivity {
    private TextView resultTextView2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_display_message);

        // Get the Intent that started this activity and extract the string
        Intent intent = getIntent();
        String bmiMessage = intent.getStringExtra(SecondaryActivity.EXTRA_MESSAGE);

        // Capture the layout's TextView and set the string as its text
        resultTextView2 = (TextView) findViewById(R.id.resultTextView);
        resultTextView2.setText(bmiMessage);
    }
}

File: LAB4/app/src/main/java/com/example/bmicalculator/DisplayMessageActivity.kt
package com.example.bmicalculator

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class DisplayMessageActivity : AppCompatActivity() {
    private var resultTextView2: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_display_message)

        // Get the Intent that started this activity and extract the string
        val intent = intent
        val bmiMessage = intent.getStringExtra(SecondaryActivity.Companion.EXTRA_MESSAGE)

        // Capture the layout's TextView and set the string as its text
        resultTextView2 = findViewById(R.id.resultTextView)
        resultTextView2!!.setText(bmiMessage)
    }
}